My Site: Fix navbar back transition animation

hassaanelgarem opened the pull request:
Fixes #18039

## Description

The navigation bar is distorted during the animation back to MySite from Stats, Activity Log, and many other screens. This is because the space behind an opaque navbar is empty so when the navbar disappears during the transition, the view behind it becomes visible.

## Solution

Setting `extendedLayoutIncludesOpaqueBars` to true means now the view of the view controller extends behind the navbar filling behind it. So when the navbar disappears the view is still showing behind it.  

## Testing Instructions
1. Go to MySIte 
2. Scroll down a bit
3. Go to Stats
4. Navigate back slowly by swiping left to right
5. ✅ Observe that the transition animation within the navbar is now smooth.
6. Repeat the above steps for all possible segues inside My Site.

Please test this in light and dark modes.

https://user-images.githubusercontent.com/25306722/155916141-106faf1f-5371-4c8b-a1fa-aefc0624da29.MP4

https://user-images.githubusercontent.com/25306722/155916167-4bcf6f93-c432-447f-895b-0daefe0db06c.MP4

## Regression Notes
1. Potential unintended areas of impact
The appearance of all the VCs affected.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manually tested all VCs in light and dark mode.

3. What automated tests I added (or what prevented me from doing so)
N/A

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

hassaanelgarem committed:
Fix navigating back from stats screen

hassaanelgarem committed:
Fix navigating back from Activity Log to MySite

hassaanelgarem committed:
Fix navigating back from Comments, Media and Menus to MySite

hassaanelgarem committed:
Fix navigating back from Sharing, People, and Settings to MySite

hassaanelgarem committed:
Add separating lines

hassaanelgarem committed:
Move configuration from BaseActivityListVC to JetpackActivityLogVC

momo-ozawa reviewed (commented):
@hassaanelgarem 👋 

There are some other VCs that need to be configured:
- `BackupListviewController`
- `JetpackScanViewController`
- `JetpackSettingsViewController`
- `SharingButtonsViewController`

I searched for `extendedLayoutIncludesOpaqueBars` in the project and compared the VCs we're showing from `BlogDetailsViewController`.

hassaanelgarem committed:
Configure other VCs to extend behind opaque nav bars
BackupListVC, JetpackScanVC, JetpackSettingsVC, SharingButtonsVC, DomainsDashboardView, BlogDashboardVC, PlanListVC

hassaanelgarem commented:
@momo-ozawa I configured the VCs you mentioned, in addition to these as well:
* `DomainsDashboardView`: SwiftUI view presented from My Site (looks disabled now by a feature flag but thought I should handle it either way)
* `BlogDashboardVC`: In a specific case it can be presented from MySite using a segue. (Configured this only in case it's pushed onto the navigation stack, to not impact the view when displayed using segmented control)
* `PlanListVC`: Is disabled right now (commented out) but thought I should handle it either way
